module/lua/setjmp/setjmp_sparc.S

```diff
@@ -0,0 +1,87 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 1992, 2010, Oracle and/or its affiliates. All rights reserved.
+ */
+
+
+#if defined(__sparcv9)
```

Comment:
This won't work as ```__sparcv9``` isn't available on Linux but on Solaris only.

The portable version to differentiate between 32 and 64 bits would be:

```
#if defined(__sparc__) && (__arch64__)
// 64 bits
#elif defined (__sparc__)
// 32 bits
#endif
```

The compiler definitions on Linux are (the -m64 is default on Linux/sparc64):

```
root@deb4g:~# echo | gcc -E -dM -m64 - |grep sparc
#define sparc 1
#define __sparc__ 1
#define __sparc 1
#define __sparc_v9__ 1
root@deb4g:~# echo | gcc -E -dM -m64 - |grep arch
#define __arch64__ 1
root@deb4g:~#
```

and on Solaris:

```
sysadmin@deimos:~$ echo | gcc -E -dM -m64 - |grep sparc
#define sparc 1
#define __sparc__ 1
#define __sparcv9 1
#define __sparc 1
sysadmin@deimos:~$ echo | gcc -E -dM -m64 - |grep arch
#define __arch64__ 1
sysadmin@deimos:~$
````